MetaLamp/lending-pool/src/Plutus/Contracts/Core.hs

```diff
@@ -0,0 +1,137 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE DerivingStrategies    #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE RecordWildCards       #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE StandaloneDeriving    #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeFamilies          #-}
+
+module Plutus.Contracts.Core where
+
+import           Control.Monad                    hiding (fmap)
+import qualified Data.ByteString                  as BS
+import qualified Data.Map                         as Map
+import           Data.Text                        (Text, pack)
+import           Data.Void                        (Void)
+import           Ledger                           hiding (singleton)
+import           Ledger.Constraints               as Constraints
+import           Ledger.Constraints.OnChain       as Constraints
+import           Ledger.Constraints.TxConstraints as Constraints
+import qualified Ledger.Scripts                   as Scripts
+import qualified Ledger.Typed.Scripts             as Scripts
+import           Playground.Contract
+import           Plutus.Contract                  hiding (when)
+import           Plutus.V1.Ledger.Value
+import qualified PlutusTx
+import           PlutusTx.Prelude                 hiding (Semigroup (..),
+                                                   unless)
+import           Prelude                          (Semigroup (..))
+import qualified Prelude
+
+deriving anyclass instance ToSchema AssetClass
+
+newtype Aave = Aave
+    { aaveProtocolInst :: AssetClass
+    } deriving stock    (Show, Generic)
+      deriving anyclass (ToJSON, FromJSON, ToSchema)
+
+PlutusTx.makeLift ''Aave
+
+instance Prelude.Eq Aave where
+    u == v = aaveProtocolInst u Prelude.== aaveProtocolInst v
+
+instance Prelude.Ord Aave where
+    compare u v = Prelude.compare (aaveProtocolInst u) (aaveProtocolInst v)
+
+type ReserveId = AssetClass
+
+data Reserve = Reserve
+    { rCurrency       :: ReserveId,
```

Comment:
Why do we need `rCurrency`? Shouldn't reserve be uniquely identifiable by `(rAToken, rDebtToken)`